modules/pipeline/providers/queuemanager/interface.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package queuemanager
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/erda-project/erda-proto-go/core/pipeline/queue/pb"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/pipeline/dbclient"
+)
+
+type Interface interface {
+	DistributedHandleIncomingPipeline(ctx context.Context, pipelineID uint64)
+	DistributedStopPipeline(ctx context.Context, pipelineID uint64)
+	DistributedQueryQueueUsage(ctx context.Context, queue *apistructs.PipelineQueue) *pb.QueueUsage
+	DistributedUpdateQueue(ctx context.Context, queueID uint64)
+	DistributedBatchUpdatePipelinePriority(ctx context.Context, queueID uint64, pipelineIDsOrderByPriorityFromHighToLow []uint64)
+}
+
+func (q *provider) DistributedHandleIncomingPipeline(ctx context.Context, pipelineID uint64) {
+	var wg sync.WaitGroup
```

Comment:
too complex func, can you separate to some clear meaning small funcs?

Reply:
yes, splitted.